scripts/katana_version/__main__.py

```diff
@@ -165,20 +167,53 @@ def provenance_subcommand(args):
     except CalledProcessError:
         pass
 
-    environment_to_capture = {"CONDA_PREFIX"}
+    try:
+        values.update(lsb_codename=capture_command("lsb_release", "-cs"))
+    except CalledProcessError:
+        pass
+
+    environment_to_capture = {
+        re.compile(r"CONDA_.*"),
+        "CXX",
+        "CC",
+        "LD",
+        "CXXFLAGS",
+        "CFLAGS",
+        "LDFLAGS",
+        "SHELL",
+        re.compile(r"CMAKE_.*"),
+        re.compile(r".*PATH"),
+    }
 
     for var in environment_to_capture:
-        if var in environ:
-            values[var.lower()] = environ[var]
+        if isinstance(var, re.Pattern):
+            vars = [k for k in environ.keys() if var.fullmatch(k)]
+        else:
+            vars = [var]
```

Comment:
This might be too paranoid so feel free to ignore, but perhaps we can have a defensive check here to make sure we definitely do not include any environment variable that has `(?![a-z0-9])key` (or something) in its name.